modules/internal/ChapelArray.chpl

```diff
@@ -1108,16 +1108,38 @@ module ChapelArray {
       compilerError("associative domains do not support .stridable");
     }
 
-    pragma "no doc"
-    inline proc these() {
-      return _value.these();
+    iter these() {
+      for i in _value.these() {
+        yield i;
+      }
+    }
+    iter these(param tag: iterKind)
+      where tag == iterKind.standalone &&
+            __primitive("method call resolves", _value, "these", tag=tag) {
+      for i in _value.these(tag) do
+        yield i;
+    }
+    iter these(param tag: iterKind)
+      where tag == iterKind.leader {
+      // If I use forall here, it says
+      //   error: user invocation of a parallel iterator should not supply tag
```

Comment:
@ronawho - I agree that is what the compiler would do with it now. I'm saying that *as a language design matter* `forall` makes more sense to me here because it includes the fact that parallelism is generated. I believe we could, if we wanted to, make the compiler treat a 'forall' in this situation the same way it currently treats a 'for'.